common/lib/util/eventemitter.ts

```diff
@@ -109,18 +111,17 @@ var EventEmitter = (function() {
 		}
 
 		if(Utils.isArray(event)) {
-			var self = this;
-			Utils.arrForEach(event, function(ev) {
-				self.off(ev, listener);
+			event.forEach((ev) => {
+				this.off(ev, listener);
 			});
 		}
 
 		/* "normal" case where event is an actual event */
 		if(listener) {
-			removeListener([this.events, this.eventsOnce], listener, event);
+			removeListener([this.events, this.eventsOnce], listener, event as string);
 		} else {
-			delete this.events[event];
-			delete this.eventsOnce[event];
+			delete this.events[event as string];
```

Comment:
The use of [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) feels like a red flag here. Are you not able to provide TypeScript with enough information by this point not to need this?

Reply:
I agree with Quintin, a better way to convince ts that if we get this far then event must be a string is just adding a runtime check before the 'event is string' section starts, eg
```
if(typeof event !== 'string') {
	throw new Error('EventListener.off: invalid function arguments ' + Utils.inspect(arguments));
}
```
that'll let you remove the assertion